src/main/kotlin/com/exactpro/th2/kafka/client/MessageProcessor.kt

```diff
@@ -29,19 +31,22 @@ import io.reactivex.rxjava3.processors.UnicastProcessor
 import io.reactivex.rxjava3.subscribers.DisposableSubscriber
 import mu.KotlinLogging
 import java.io.Closeable
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicLong
 import java.util.function.Supplier
 
 class MessageProcessor(
     rawSubscriberFactory: Supplier<MessageRouterSubscriber<RawMessageBatch>>,
     config: Config
 ) : Closeable {
-    private val processor: FlowableProcessor<RawMessage> = UnicastProcessor.create()
+    private val processor: FlowableProcessor<RawMessage.Builder> = UnicastProcessor.create()
 
     init {
         createPipeline().subscribe(TerminationSubscriber(rawSubscriberFactory, config))
     }
 
-    fun process(message: RawMessage) = processor.onNext(message)
+    fun process(message: RawMessage.Builder) = processor.onNext(message)
```

Comment:
If we call it from different threads we probably should use synchronization. UnicastProcessor uses single-producer-single-consumer queue under the hood. It means we need to synchronize our access to the processor